x-pack/plugins/embeddable_enhanced/public/embeddables/dynamic_action_storage.test.ts

```diff
@@ -5,75 +5,67 @@
  * 2.0.
  */
 
-import {
-  Embeddable,
-  ViewMode,
-  VALUE_CLICK_TRIGGER,
-  SELECT_RANGE_TRIGGER,
-} from '@kbn/embeddable-plugin/public';
-import {
-  EmbeddableActionStorage,
-  EmbeddableWithDynamicActionsInput,
-} from './embeddable_action_storage';
-import { UiActionsEnhancedSerializedEvent } from '@kbn/ui-actions-enhanced-plugin/public';
+import { SELECT_RANGE_TRIGGER, VALUE_CLICK_TRIGGER } from '@kbn/embeddable-plugin/public';
 import { of } from '@kbn/kibana-utils-plugin/public';
+import { UiActionsEnhancedSerializedEvent } from '@kbn/ui-actions-enhanced-plugin/public';
+import { BehaviorSubject } from 'rxjs';
+import { DynamicActionStorage, DynamicActionStorageApi } from './dynamic_action_storage';
 // use real const to make test fail in case someone accidentally changes it
 import { APPLY_FILTER_TRIGGER } from '@kbn/data-plugin/public';
+import { DynamicActionsSerializedState } from '../plugin';
 
-class TestEmbeddable extends Embeddable<EmbeddableWithDynamicActionsInput> {
-  public readonly type = 'test';
-  constructor() {
-    super({ id: 'test', viewMode: ViewMode.VIEW }, {});
-  }
-  reload() {}
-}
+const getApi = (): DynamicActionStorageApi => {
+  const dynamicActionsState$ = new BehaviorSubject<DynamicActionsSerializedState['enhancements']>({
+    dynamicActions: { events: [] },
+  });
+  return {
+    setDynamicActions: (newDynamicActions) => {
+      dynamicActionsState$.next(newDynamicActions);
+    },
+    dynamicActionsState$,
+  };
+};
 
 describe('EmbeddableActionStorage', () => {
   describe('.create()', () => {
     test('method exists', () => {
-      const embeddable = new TestEmbeddable();
-      const storage = new EmbeddableActionStorage(embeddable);
+      const storage = new DynamicActionStorage('testId', () => 'testTitle', getApi());
       expect(typeof storage.create).toBe('function');
     });
 
     test('can add event to embeddable', async () => {
-      const embeddable = new TestEmbeddable();
-      const storage = new EmbeddableActionStorage(embeddable);
+      const api = getApi();
+      const storage = new DynamicActionStorage('testId', () => 'testTitle', api);
       const event: UiActionsEnhancedSerializedEvent = {
         eventId: 'EVENT_ID',
         triggers: ['TRIGGER-ID'],
         action: {} as any,
       };
 
-      const events1 = embeddable.getInput().enhancements?.dynamicActions?.events || [];
+      const events1 = api.dynamicActionsState$.getValue()?.dynamicActions.events ?? [];
       expect(events1).toEqual([]);
 
+      const spy = jest.spyOn(api, 'setDynamicActions');
       await storage.create(event);
+      expect(spy).toBeCalledWith({
+        dynamicActions: {
+          events: [
+            {
+              eventId: 'EVENT_ID',
+              triggers: ['TRIGGER-ID'],
+              action: {} as any,
```

Comment:
type `as unknown as SerializedAction`

Reply:
Done in 343e464a979251cae24177f87eb348c52ab673b0